File: pubnub-java/src/main/java/com/pubnub/api/callbacks/SubscribeCallback.java
package com.pubnub.api.callbacks;

import com.pubnub.api.PubNub;
import com.pubnub.api.models.consumer.PNStatus;
import com.pubnub.api.v2.callbacks.EventListener;
import com.pubnub.api.v2.callbacks.StatusListener;
import org.jetbrains.annotations.NotNull;

public abstract class SubscribeCallback implements StatusListener, EventListener {
    public static class BaseSubscribeCallback extends SubscribeCallback {
        @Override
        public void status(@NotNull PubNub pubnub, @NotNull PNStatus pnStatus) {
        }
    }
}

File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/callbacks/SubscribeCallback.kt
package com.pubnub.api.java.callbacks

import com.pubnub.api.java.PubNub
import com.pubnub.api.java.v2.callbacks.EventListener
import com.pubnub.api.java.v2.callbacks.StatusListener
import com.pubnub.api.models.consumer.PNStatus

abstract class SubscribeCallback : StatusListener, EventListener {
    open class BaseSubscribeCallback : SubscribeCallback(), EventListener {
        override fun status(pubnub: PubNub, pnStatus: PNStatus) {
        }
    }
}
